### DISCUSSION TITLE
Creating convenience function for code block of specific language

### DISCUSSION THREAD

**Discussion by ljarosch** [typst/typst#3540] [Q&A] [answered]
Hi, I currently have to write a lot of code-blocks like this (ignore the comment I just added it so that markdown renders the nested block correctly):

```typst
```python
print("Hello")
print("World")
```//ignore
```

I was wondering if it would be possible to define a convenience function that allows me to do this instead:
```typst
#python[
    print("Hello")
    print("World")
]
```

This would speed up the typing a lot, however I was struggling to implement this because `#raw()` expects a string input and converting content to strings seems pretty complicated, especially if I want it to be newline-sensitive like in my example. I am still fairly new to Typst, so not sure if something like that is really possible. Would appreciate any tips!

**Comment by Enter-tainer** [comment]
https://typst.app/docs/reference/text/raw/

raw has a special markup syntax

**Reply by Enter-tainer** [reply]
tldr, simply use the markdown syntax

**Comment by emilyyyylime** [comment]
If you really want to save on characters I think using single \`s would also work

**Comment by A-Walrus** [comment, marked as answer]
If all/most of the code blocks in your document are of the same language, maybe something like this would be better:
```typ
#set raw(lang: "python")
```
If you really want it to be a function you could do something like:
```typ
#let python(r) = {
  set raw(lang:"python")
  r
}

#python[`print("hello world")`]

```

also like @emilyyyylime said, you could use single `s to save some chars.

**Comment by ljarosch** [comment]
Thanks @A-Walrus and @emilyyyylime, that's close enough to what I was trying
